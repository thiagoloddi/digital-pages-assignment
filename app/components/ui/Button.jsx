import React from 'react';
import injectSheet from 'react-jss';

const buttonStyle = {
  button: props => ({
      color: 'white',
      border: 'none',
      outline: 'none',
      padding: '15px 50px',
      cursor: 'pointer',
      backgroundColor: '#00ACED',
      fontWeight: 'bold',
      borderRadius: '10px',
      minWidth: '100px',
      fontSize: '20px',
      // float: props.right ? 'right' : ''
  })
};

const Button = ({ label, onClick, className, classes, type="button" }) => {
  return (
    <button onClick={onClick} className={`${classes.button} ${className}`} type={type}>
      {label}
    </button>
  );
}

export default injectSheet(buttonStyle)(Button);