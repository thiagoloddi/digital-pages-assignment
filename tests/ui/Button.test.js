import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../app/components/ui/Button';

const func = () => {};

it('renders correctly', () => {
  const tree = renderer
    .create(<Button label="some label" onClick={func} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});