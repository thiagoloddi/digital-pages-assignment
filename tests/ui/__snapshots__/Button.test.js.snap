// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="Button-button-0-1-1 Button-button-0-1-2 undefined"
  onClick={[Function]}
  type="button"
>
  some label
</button>
`;
